src/test/java/se/sundsvall/messaging/integration/db/HistoryRepositoryTest.java

```diff
@@ -0,0 +1,155 @@
+package se.sundsvall.messaging.integration.db;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.groups.Tuple.tuple;
+import static org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace.NONE;
+import static se.sundsvall.messaging.model.MessageType.SNAIL_MAIL;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.test.context.jdbc.Sql;
+
+import se.sundsvall.messaging.integration.db.entity.HistoryEntity;
+import se.sundsvall.messaging.test.annotation.UnitTest;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = NONE)
+@UnitTest
+@Sql(scripts = {
+	"/db/scripts/truncate.sql",
+	"/db/scripts/testdata-ut.sql"
+})
+class HistoryRepositoryTest {
+
+	@Autowired
+	private HistoryRepository historyRepository;
+
+	@Test
+	void findByMunicipalityIdAndDeliveryId() {
+		// Arrange
+		final var municipalityId = "2281";
+		final var deliveryId = "ea6b0684-69d5-4f70-8b2d-6255303ead0e";
+
+		// Act
+		final var match = historyRepository.findByMunicipalityIdAndDeliveryId(municipalityId, deliveryId);
+
+		// Assert
+		assertThat(match).isPresent().map(entity -> {
```

Comment:
Should probably be using .satisfies() instead of map here.

Reply:
As match is an Optional<>, satisfies doesn't work here. You could separate the test to:
```
		assertThat(match).isPresent();
		match.ifPresent(entity -> {
			assertThat(entity.getMunicipalityId()).isEqualTo(municipalityId);
			assertThat(entity.getDeliveryId()).isEqualTo(deliveryId);
			assertThat(entity.getBatchId()).isEqualTo("cb1af665-835f-45b8-8755-9aa2ed284292");
			assertThat(entity.getMessageId()).isEqualTo("d5161acb-2462-4065-a679-53b1cd77be92");
			assertThat(entity.getMessageType()).isEqualTo(SNAIL_MAIL);
			assertThat(entity.getOrigin()).isEqualTo("origin1");
			assertThat(entity.getIssuer()).isEqualTo("issuer1");
		});
```
but then again, the map works as well.

Reply:
Yes but since m in that case is an Optional<> you need to do m.get().methodName() each time instead of just m.methodName() (which imo is ugglier).

Reply:
I found an even better solution (hasValueSatisfying) that I will update test with:
```
		// Assert
		assertThat(match).isPresent().hasValueSatisfying(entity -> {
			assertThat(entity.getMunicipalityId()).isEqualTo(municipalityId);
			assertThat(entity.getDeliveryId()).isEqualTo(deliveryId);
			assertThat(entity.getBatchId()).isEqualTo("cb1af665-835f-45b8-8755-9aa2ed284292");
			assertThat(entity.getMessageId()).isEqualTo("d5161acb-2462-4065-a679-53b1cd77be92");
			assertThat(entity.getMessageType()).isEqualTo(SNAIL_MAIL);
			assertThat(entity.getOrigin()).isEqualTo("origin1");
			assertThat(entity.getIssuer()).isEqualTo("issuer1");
		});
```
this is the way to go 😀!